### PR TITLE
layout complete, lacking functions

### PR DESCRIPTION
**Pull request by markellisdev**
This is what I have so far. I still need to build a function and add event listeners. Am I on the right track?

### CODE REVIEW

**Comment by markellisdev** [comment]
I'm now on the final function to build my tree. Thinking through how to tell it to print number of spaces. I know how to calculate the math, just not sure yet how to tell it print " " \* number...

File: quiz.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+
+
+// Once the user enters in a number, and a character, the user can either then just press the enter key (as long as the cursor is in one of the input fields), or click a button that is labeled "Grow your tree" and the tree should be shown in the console. This requires you to add an event listener to the button, as well as an event listener for the enter/return key.
+
+
+treeButton.addEventListener("click", checkInput);
+
+userChar.addEventListener("submit", checkInput);
```
Comment (gregkorte) [inline quiz.js]: When you tackle the `Enter` key press event, this listener will probably change.

File: quiz.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+userHeight.addEventListener("submit", checkInput);
+
+
```
Comment (gregkorte) [inline quiz.js]: Try to have a consistent spacing throughout your files.

File: quiz.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	alert("Please make sure you enter a number for height and a character before trying to grow your tree");
+// If either of the input fields does not have a value in it when the user presses the enter key, or presses the button, then display an alert stating that both fields must have a value.
+	}
+	else {
+		growTree(height, char);
+	}
+}
+
+
+function growTree(height, char) {
```
Comment (gregkorte) [inline quiz.js]: Here you are passing the properties of the object. Instead you should be passing the entire object.

File: quiz.js [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+// A key that specifies the height of the pine tree.
+
+// The value for the height of the tree should be from user input in a <input type="text"> field in the DOM.
+
+// A key that specifies which character to use to build the pine tree.
+
+// The character to use should be from user input in a <input type="text"> field in the DOM.
+
+
+var tree = {
```
Comment (gregkorte) [inline quiz.js]: Are the properties in this object ever set?

File: quiz.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+	height: userHeight,
+	character: userChar
+};
+
+
+// function growTree(height, char) {
```
Comment (gregkorte) [inline quiz.js]: I know this is incomplete as of now, but in final submission remember to clean up any commented code to keep your files as readable and easy to follow as possible.

File: quiz.js [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+var treeButton = document.getElementById("yourTree");
+
+
+
+
+function checkInput(yes) {
+	var height = document.getElementById("userHeight").value;
+	var char = document.getElementById("userChar").value;
+	if (isNaN(height)  || (char.length > 1 || char.length < 1)) {
+	alert("Please make sure you enter a number for height and a character before trying to grow your tree");
```
Comment (gregkorte) [inline quiz.js:32]: Watch your indentation.

**Comment by gregkorte** [comment]
Nice start on your quiz, Mark! Please see the above comments, make your changes and additions, and let me know via slack when you are finished. I'll take another look.

File: quiz.js [hunk 7]
```diff
@@ -10,66 +9,53 @@
 
 // The character to use should be from user input in a <input type="text"> field in the DOM.
 
+// It accepts a single object as an argument. The object should have two key/value pairs.
 
 var tree = {
 	height: userHeight,
```
Comment (gregkorte) [inline quiz.js]: Is it necessary to create these properties or will they be created when you actually have data in your `checkInput` function?

File: quiz.js [hunk 8]
```diff
@@ -10,66 +9,53 @@
...
-
-
-
 var treeButton = document.getElementById("yourTree");
 
+//This function is for the Enter button
 
+function enterInput(event) {
+	if (event.keyCode === 13) {
+		checkInput(true);
```
Comment (gregkorte) [inline quiz.js]: What's this `true`?

File: quiz.js [hunk 9]
```diff
@@ -10,66 +9,53 @@
...
+	}
+};
 
+// If either of the input fields does not have a value in it when the user presses the enter key, or presses the button, then display an alert stating that both fields must have a value.
 
 function checkInput(yes) {
```
Comment (gregkorte) [inline quiz.js]: What's this `yes`?

**Comment by gregkorte** [comment]
Nice work, Mark. Just a few comments left. Let me know via slack when you're ready for me to take another look.

File: quiz.js (hunk 7 above)
Comment (markellisdev) [reply quiz.js]: I see I'm creating them inside the function, so I removed them.

File: quiz.js (hunk 9 above)
Comment (markellisdev) [reply quiz.js]: For some reason I thought I needed to pass a value into the function, but I see I didn't use it anywhere inside, so I removed it.

File: quiz.js (hunk 8 above)
Comment (markellisdev) [reply quiz.js]: Looking back, I'm not sure if I thought I needed to pass a value through to checkInput.

**Comment by gregkorte** [comment]
You are good to merge, Mark! :+1:

**Comment by markellisdev** [comment]
Thank you Greg!

**Pull request merged**